staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/matching/matching_test.go

```diff
@@ -424,11 +426,38 @@ func TestMatcher(t *testing.T) {
 			attrs:         admission.NewAttributesRecord(nil, nil, schema.GroupVersionKind{"autoscaling", "v1", "Scale"}, "ns", "name", schema.GroupVersionResource{"extensions", "v1beta1", "deployments"}, "", admission.Create, &metav1.CreateOptions{}, false, nil),
 			expectMatches: false,
 		},
+		{
+			name: "bindingMatch treat empty ResourceRules as match",
+			criteria: &v1alpha1.MatchResources{
+				NamespaceSelector: &metav1.LabelSelector{},
+				ObjectSelector:    &metav1.LabelSelector{},
+				ExcludeResourceRules: []v1alpha1.NamedRuleWithOperations{{
+					RuleWithOperations: v1alpha1.RuleWithOperations{
+						Operations: []v1.OperationType{"*"},
+						Rule:       v1.Rule{APIGroups: []string{"extensions"}, APIVersions: []string{"v1beta1"}, Resources: []string{"deployments"}, Scope: &allScopes},
+					},
+				}},
+			},
+			attrs:         admission.NewAttributesRecord(nil, nil, schema.GroupVersionKind{"autoscaling", "v1", "Scale"}, "ns", "name", schema.GroupVersionResource{"apps", "v1", "deployments"}, "", admission.Create, &metav1.CreateOptions{}, false, nil),
+			expectMatches: true,
+			isBinding:     true,
+		},
+		{
+			name: "bindingMatch treat non-empty ResourceRules as no match",
```

Comment:
This is a really interesting case. Should we even allow resourceRules to be set but empty? This effectively disables the entire binding. Do we want that? Would it be better to disallow this case and instead ask users to delete the binding if they don't need it?

Reply:
Sorry I should delete this test case since we have added into validation that prevent this from happening. Will update shortly

Reply:
I can live with an empty list meaning "match nothing" to be consistent.

Reply:
now an empty list means "match everything", but definitions must have at least one rule (enforced via validation).